modules/cli/src/main/scala/scala/cli/ScalaCli.scala

```diff
@@ -70,7 +70,23 @@ object ScalaCli extends CommandsEntryPoint {
     }
   }
 
+  private def partitionArgs(args: Array[String]): (Array[String], Array[String]) = {
+    val systemProps = args.takeWhile(_.startsWith("-D"))
+    (systemProps, args.drop(systemProps.size))
+  }
+
+  private def setSystemProps(systemProps: Array[String]): Unit = {
+    systemProps.map(_.stripPrefix("-D")).foreach { prop =>
+      prop.split("=", 2) match {
+        case Array(key, value) =>
+          System.setProperty(key, value.stripPrefix("=")) // todo log
```

Comment:
We should handle the `Array(key)` case too, corresponding to arguments like `-Dfoo` (which are legit). Also, we don't need the `stripPrefix("=")`, `split` should already have removed it.
```suggestion
        case Array(key, value) =>
          System.setProperty(key, value)
        case Array(key) =>
          System.setProperty(key, "")
```